### DISCUSSION TITLE
Change shape of clicked element

### DISCUSSION THREAD

**Discussion by TobiasRama** [clientIO/joint#1908] [Q&A] [answered]
Is it possible to change an elements shape by clicking a custom button? Like changing a rectangle to a circle and the other way. Or do I have to remove the rectangle and create a new circle element?

**Comment by kumilingus** [comment, marked as answer]
- You can use `<path>` and change its `d` attribute (path is the most general shape). See [FTA demo](https://resources.jointjs.com/demos/fta).

- You can change the SVG markup dynamically.
```js
el.set('markup', joint.util.svg`<rect @selector="body" />`);
el.set('markup', joint.util.svg`<ellipse @selector="body" />`);
```
- Yes, you can replace one element with another - See [CodePen](https://codepen.io/jointjs/pen/BaYOREQ/597a7a6c98f880bd89fc441826076722).

- You can have a custom view, to render complex shapes changing their look based on the model's attributes
  - https://resources.jointjs.com/tutorial/content-driven-element
  - https://jsfiddle.net/kumilingus/gdks4tc9/6/

It depends. If you need to change just the shape (morphology) of a single `SVGElement`, I'd recommend using `SVGPath`.

**Reply by TobiasRama** [reply]
That was helpful, thank you very much!
